### PR TITLE
[ElectedRepresentative][ADMIN] Add a zone filter

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

**Comment by illusionOfParadise** [comment]
[ch11499]

File: src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php [hunk 1]
```diff
@@ -426,6 +429,84 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
                     return true;
                 },
             ])
+            ->add('mandates.zone', CallbackFilter::class, [
+                'label' => 'Périmètres géographiques',
+                'show_filter' => true,
+                'field_type' => ModelAutocompleteType::class,
```
Comment (ottaviano) [inline src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php:435]: maybe we could use `ModelAutocompleteFilter` class instead?

File: src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php [hunk 2]
```diff
@@ -426,6 +429,84 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
...
+                            case ZoneCategory::DEPARTMENT:
+                                if (!\in_array('category', $qb->getAllAliases(), true)) {
+                                    $qb->leftJoin('zone.category', 'category');
+                                }
+
+                                if (!\in_array('referentTag', $qb->getAllAliases(), true)) {
+                                    $qb->leftJoin('zone.referentTags', 'referentTag');
+                                }
+
+                                foreach ($zone->getReferentTags() as $key_t => $referentTag) {
```
Comment (ottaviano) [inline src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php:487]: will it be enough? 
```php
->andWhere('tags IN (tags) AND category != region')
``` 
?

File: src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php (hunk 1 above)
Comment (illusionOfParadise) [reply src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php:435]: It's not possible, the class should implement `FormTypeInterface`, that is not the case of `ModelAutocompleteFilter`.

**Pull request merged**